library/std/src/io/copy.rs

```diff
@@ -71,18 +104,67 @@ where
     //     based on its privileged knowledge of unstable rustc
     //     internals;
     unsafe {
-        reader.initializer().initialize(buf.assume_init_mut());
+        reader.initializer().initialize(MaybeUninit::slice_assume_init_mut(buf));
```

Comment:
This requires attention since I don't know if those changes are still adhere to those "unstable rustc internals" mentioned above.